pipeline/src/node_properties/video_source_properties.rs

```diff
@@ -0,0 +1,92 @@
+use crate::resolution::Resolution;
+use serde::{Deserialize, Serialize};
+use url::Url;
+use uuid::Uuid;
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(tag = "source_type", rename_all = "snake_case")]
```

Comment:
Very glad that this change didn't mean we had to go back to manual Deserialize impl.